docs/build-mvm/glossary.md

```diff
@@ -1,257 +1,419 @@
 
-## Subnetwork
+> TODO See this repo for more.
 
-A Subnetwork is a virtual partition (overlay) of the [Fluence p2p network](#fluence-network) defined by the terms and life of a Deal and accessed and managed with [Aqua](#Aqua). The behavior of a Subnetwork is defined by the set of [Worker](#worker)s specified by the [Deal](#deal).  
+### Init Peer
 
-Subnetworks are an integral part of the [Fluence protocol](#fluence-protocol) and offer additional qualities, like fault tolerance or read consensus. That depends on the [Aqua](#Aqua) code that is deployed on the Subnet [Worker](#worker)s to maintain emergent qualities, and on the [Aqua](#Aqua) code that is used to work with the subnet.
+The [Peer](#peer) that initiates a [Cloudless Function](#cloudless-function) – creates a [Particle](#particle) and sends it over [Fluence Protocol](#fluence-protocol).
 
-To use Subnet, a developer needs to provide the [Deal](#deal) ID and possibly some additional authorization information.
+[Init Peer ID](/docs/aqua-book/language/topology.md#init_peer_id) can always be accessed from [Aqua](#Aqua) or via [Marine SDK](#marine-sdk).
 
-## Spell
+### Client (peer)
 
-Spell is an [AIR](#air) script hosted on a [Peer](#peer) that is triggered periodically, similar to a cron job, or by an external event.
+A Fluence [Peer](#peer) connected to a Relay, but not participating in the Fluence Kademlia network – so not a [Host](#host). Most likely a running [Fluence JS Client](#fluence-js-client) (browser, CLI).
 
-Spell has its own small KV store and a mailbox, that makes it very close to the Actor model.
+Client peers, as well as [Worker](#worker)s, cannot be discovered using their respective [PeerId](#peerid)s. Developer needs to discover what [Host](#host) or [Relay](#relay) is advertized to Kademlia network, and use [Aqua](#aqua) to ensure the right path is used.
 
-## Tetraplet
+### Relay
 
-Tetraplet is a data structure describing the origin of an argument to a function call. It contains four fields:
+A Fluence Peer, a [Host](#host), that participates in [Cloudless Function](#cloudless-function)s by providing its advanced connectivity capabilities, which includes:
 
-- `peer_id`
-- `service_id`
-- `function_name`
-- `functor`
+- Ability of a Host to send messages internally to hosted [Worker](#worker)s
+- Ability of a Host to send messages to connected [Client Peer](#client-peer)s
 
-Tetraplets are the mean to check and enforce distributed security invariants of a function by putting restrictions on its arguments, e.g.: the argument “is_authorized” must be provided by a function “check_auth” from “security” [Service](#service) run on the same [Peer](#peer). Read more about Security Tetraplets [here](/docs/build-mvm/security.md).
+To execute code on a [Peer](#peer) that's behind a Relay, [use `on peer via relay:` construct](/docs/aqua-book/language/topology#accessing-peers-via-other-peers) in [Aqua](#Aqua).
```

Comment:
```suggestion
To execute code on a [Peer](#peer) that's behind a Relay, e.g., behind a NAT, use the [`on peer via relay:`](/docs/aqua-book/language/topology#accessing-peers-via-other-peers) construct available in [Aqua](#Aqua).
```

Reply:
The most common thing here is not about NAT, but about web browsers and workers.